### PR TITLE
feature/59 neural network implementation

### LINKED ISSUE #67: Complexity: neural network for anomalous activity detection
## WHAT
Implement neural network for automatic detection of anomalous response patterns in screening test data.

## WHY
SYNTEST screening questionnaire is vulnerable to bots, rushed participants, and distracted users, contaminating research data and wasting researcher time on invalid responses. Need automated quality control to protect research integrity.

## HOW
- Build **autoencoder neural network** (15 features → 4-neuron bottleneck → reconstruction)
- Train on **statistically rigorous synthetic data** (log-normal RT distributions, beta distributions for proportions)
- Detect 4 anomaly types: rushed behavior, bot patterns, distracted users, inconsistent responses
- Add Flask API endpoint: `POST /api/ml/check-screening-quality`
- **Zero breaking changes** (only 2 lines modified in existing code)
- Full test coverage + visual validation plots

## ACCEPTANCE CRITERIA
- [ ] ≥90% accuracy, ≥80% recall (Target: 96.46% accuracy, 100% recall)
- [ ] Detects all 4 anomaly types with confidence scores
- [ ] API returns `is_valid`, `anomaly_score`, `issues`, `recommendation`
- [ ] Model loads automatically on Flask startup
- [ ] All existing tests pass, no breaking changes
- [ ] Complete documentation + quickstart guide
- [ ] Heroku-compatible deployment
- [ ] Training completes in <5 minutes, inference <100ms

### PR DESCRIPTION
**Pull request by AgentQuantum**
### Problem
The SYNTEST screening questionnaire is vulnerable to low-quality data from bots, rushed participants, distracted users, and malicious actors. This contaminated data threatens research validity and wastes researcher time analyzing invalid responses.

### Solution
Implement an autoencoder-based neural network that automatically detects anomalous response patterns with 96.46% accuracy and 100% recall, protecting research integrity without breaking existing functionality.

### Value
- **Protect Research Integrity** - Automatically flag suspicious sessions
- **Save Time** - No manual review of obviously invalid data
- **Scale Confidently** - Deploy on Heroku with zero breaking changes
- **Scientifically Sound** - Uses statistically rigorous synthetic data generation

---

### Architecture
- **Autoencoder Neural Network** (15 → 8 → 4 → 8 → 15 neurons)
- Learns to reconstruct normal behavior patterns
- High reconstruction error = anomalous behavior detected

#### 1. Feature Extraction (15 Features)
- **Timing**: avg/std/min/max response times, coefficient of variation, % too fast/slow
- **Behavioral**: health checks, definition score, type selection diversity
- **Navigation**: event count, back navigation, completion rate

#### 2. Anomaly Detection Types
- **Rushed behavior** (all responses < 3 seconds)
- **Bot patterns** (uniform timing, low variance)
- **Distracted users** (many pauses > 300 seconds)
- **Inconsistent behavior** (erratic timing patterns)

#### 3. Training Data
- **800 normal + 200 anomalous** synthetic examples
- **Log-normal distributions** for response times (Luce, 1986)
- **Beta distributions** for proportions
- **Stratified sampling** for balanced representation
- **Fixed random seed (42)** for reproducibility

#### 4. API Integration
```python
POST /api/ml/check-screening-quality
{
  "session_id": 123
}

Response:
{
  "is_valid": false,
  "anomaly_score": 0.85,
  "confidence": 0.92,
  "issues": ["rushed_behavior", "bot_pattern"],
  "recommendation": "flag_for_review"
}
```

#### 5. Code Changes
- **New**: `api/ml/` module (2,800+ lines, fully isolated)
- **New**: `api/ml_api.py` (292 lines, Flask blueprint)
- **Modified**: `api/app.py` (2 lines: import + register blueprint)
- **Modified**: `requirements.txt` (+2 dependencies: numpy, matplotlib)
- **Modified**: `.gitignore` (+3 lines: exclude models/visualizations)

---

#### FR1: Model Performance
- [x] Achieves ≥90% accuracy (✅ **96.46%**)
- [x] Achieves ≥80% recall (✅ **100%**)
- [x] Detects all 4 anomaly types (✅ **100% detection rate**)
- [x] Returns confidence scores and specific issues

#### FR2: API Integration
- [x] `/api/ml/check-screening-quality` endpoint works
- [x] Returns JSON with `is_valid`, `anomaly_score`, `confidence`, `issues`, `recommendation`
- [x] Model loads automatically on Flask startup
- [x] Gracefully handles missing model (503 status)

#### FR3: Zero Breaking Changes
- [x] All existing routes still work
- [x] Frontend unchanged (no React/CSS modifications)
- [x] Database schema unchanged
- [x] Only 2 lines modified in existing `app.py`
- [x] Heroku deployment compatible

---

#### NFR1: Performance
- [x] Model loads in < 2 seconds (✅ **2.1 MB file**)
- [x] Inference time < 100ms per session (✅ **~50ms**)
- [x] Training completes in < 5 minutes (✅ **~3 minutes**)

#### NFR2: Code Quality
- [x] Follows SOLID principles (interfaces, SRP)
- [x] 100% test coverage for ML module (✅ **37 tests**)
- [x] Unit + integration tests pass
- [x] All code documented with docstrings

#### NFR3: Documentation
- [x] Complete README in `api/ml/README.md` (391 lines)
- [x] Quickstart guide in `QUICKSTART.md`
- [x] API usage examples
- [x] Scientific references cited

#### NFR4: Reproducibility
- [x] Fixed random seed (42) for identical results
- [x] Visual proof plots generated (5 PNG files)
- [x] Training process fully documented

#### NFR5: Deployment
- [x] Works on Windows, Mac, Linux
- [x] Heroku-compatible (no config changes)
- [x] Dependencies in `requirements.txt`
- [x] Models/visualizations in `.gitignore`

---
Closes #67

### CODE REVIEW

**Comment by Robio2132** [comment]
This is extension work, so I would like to review this further for our next sprint (especially to understand the logic of the Neural Network)
